showkase-processor/src/main/java/com/airbnb/android/showkase/processor/writer/ShowkaseBrowserWriter.kt

```diff
@@ -144,6 +159,44 @@ internal class ShowkaseBrowserWriter(private val environment: XProcessingEnv) {
         .addMember("numTypography = %L", typographySize)
         .build()
 
+    // This is to aggregate metadata for the custom annotation annotated with Preview
+    internal fun writeCustomAnnotationElementToMetadata(element: XElement) {
+        if (!element.isTypeElement()) return
+        if (element.isAnnotationClass() && element.qualifiedName == ShowkaseProcessor.PREVIEW_CLASS_NAME) return
```

Comment:
is there any circumstance where the element passed is `Preview`?

Reply:
I hope not, maybe I can revert this check? 🤔

Reply:
I think there is a case actually where this occurs. I think I have to keep this check.

Reply:
I think this is if you have a repeatable preview annotation. Then first when the top preview is discovered by the processor it will see that the element annotated with preview is a preview annotation and then try to run this function. But the information we want to write to the file is contained in the annotation in the next round. I think that is why we need this check somewhere in the process here or at call-site 🤔